### PR TITLE
refactor(cardinal): create wrapper functions in cardinal for component related functions.

### PR DESCRIPTION
**Pull request by pyrofolium**
Closes: #421

## What is the purpose of the change

Make public component package. Note there are temporary coding changes to get rid of circular dependencies. 

## Brief Changelog

Moved components out of ecs. 

Took GetComponent, SetComponent, Create, CreateMany and UpdateComponent and made private copies of these functions within persona. 

The objective here is to temporarily use these private functions to break the circular dependency. When we reimplement persona we can get rid of these private functions. The current version of persona is going to be deprecated anyway, so I think this pattern works. I can use it to decouple other things away from persona. 

## Testing and Verifying

All original tests still work.

### CODE REVIEW

File: cardinal/ecs/persona.go [hunk 1]
```diff
@@ -193,3 +195,112 @@ func (w *World) GetSignerForPersonaTag(personaTag string, tick uint64) (addr str
 	}
 	return addr, nil
 }
+
+// TODO private component function used to temporarily remove circular dependency until we replace components.
```
Comment (jerargus) [inline cardinal/ecs/persona.go:199]: I created a ticket to capture the plugin work: https://linear.app/arguslabs/issue/WORLD-423/ecs-plugin-feature -- please add a link to the ticket in your comment here.

**Review by jerargus — APPROVED** [review]

File: cardinal/world.go [hunk 2]
```diff
@@ -99,13 +99,13 @@ func NewMockWorld(opts ...WorldOption) (*World, error) {
 // CreateMany creates multiple entities in the world, and returns the slice of ids for the newly created
 // entities. At least 1 component must be provided.
 func (w *World) CreateMany(num int, components ...component_metadata.Component) ([]EntityID, error) {
```
Comment (namrapatel) [inline cardinal/world.go:101]: Can/should we just get rid of these duplicates?

**Review by namrapatel — COMMENTED** [review]
Are we pruning the `cardinal.World` function in a separate PR? If so, feel free to resolve, this looks good.

**Comment by pyrofolium** [comment]
@namrapatel @jerargus @technicallyty 

Ok so there was a lot of discussion. Here are the overall changes. 

There are now two packages for components in `ecs`

`component_metadata`
`components`

The reason this exists is because of the dependency graph to prevent circular dependencies. 

`world` -> `component_metadata`
`components` -> `world`

`components` was moved back into `ecs` which is "private" with the possibility of making it internal later. The reason is if I move it into `cardinal` then that forces the user to import `ecs`.

**Comment by jerargus** [comment]
Per our discussion, make sure to add the SetComponent/RemoveComponent/etc wrappers to world-engine/cardinal.

**Comment by pyrofolium** [comment]
> Per our discussion, make sure to add the SetComponent/RemoveComponent/etc wrappers to world-engine/cardinal.

Done

**Review by jerargus — APPROVED** [review]

File: cardinal/world.go (hunk 2 above)
Comment (pyrofolium) [reply cardinal/world.go]: As discussed the other Create and CreateMany in components is now being moved to components in the ecs folder.

**Pull request merged**